python/test/unit/language/test_core.py

```diff
@@ -371,6 +371,34 @@ def test_compare_op(dtype_x, dtype_y, op, mode_x, mode_y, device='cuda'):
     _test_binary(dtype_x, dtype_y, expr, numpy_expr, mode_x=mode_x, mode_y=mode_y, device=device)
 
 
+# ---------------
+# test broadcast
+# ---------------
+@pytest.mark.parametrize("dtype", dtypes_with_bfloat16 + ["*int32"])
+def test_broadcast(dtype):
+    @triton.jit
+    def broadcast_kernel(x_ptr, y_ptr, y_broadcasted_ptr, BLOCK_SIZE: tl.constexpr):
+        offset1 = tl.arange(0, BLOCK_SIZE)
+        offset2 = tl.arange(0, BLOCK_SIZE)
+        x = tl.load(x_ptr + BLOCK_SIZE * offset1[:, None] + offset2[None, :])
+        y = tl.load(y_ptr + offset2)
+        _, y_broadcasted = tl.broadcast(x, y)
+        tl.store(y_broadcasted_ptr + BLOCK_SIZE * offset1[:, None] + offset2[None, :], y_broadcasted)
+
+    SIZE = 32
```

Comment:
Can you use two different sizes here? If the matrix is square, the broadcasting test seems weaker since wrong padding will still have the right shape

Reply:
For sure, fixed.